nidirect_gp/nidirect_gp.module

```diff
@@ -246,3 +248,62 @@ function _build_gp_practice_title($practiceName = '', $surgeryName = '') {
 
   return $title;
 }
+
+/**
+ * Implements hook_views_query_alter().
+ */
+function nidirect_gp_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
+  // Alter the view code for GP Practice Lists if being sorted on the
+  // 'Online appointments' or 'Online prescriptions' fields.
+  // This is necessary because Drupal 8 Views does not seem to handle the case
+  // of sorting on a 'link' field as it does not build the join correctly
+  // and just errors on the 'unknown' field that appears in the 'orderby'.
+  if ($view->id() == 'gp_practice_list') {
+    if (isset($query->orderby)
+      && is_array($query->orderby)
+      && isset($query->orderby[0])
+      && isset($query->orderby[0]['field'])) {
+      if ($query->orderby[0]['field'] == 'unknown') {
+        // Decide whether we are sorting by the 'Online appointments' column or
+        // by 'Online precriptions'.
+        $join_table = NULL;
+        $view_input = $view->getExposedInput();
+        if (isset($view_input) && isset($view_input['order'])) {
+          if ($view_input['order'] == 'nothing') {
+            // We are sorting by 'Online appointments'.
+            $join_table = 'node__field_gp_appointments';
+          }
+          elseif ($view_input['order'] == 'nothing_1') {
```

Comment:
These unknown and nothing values that views gives really aren't terribly helpful for us, but it's done it for years so.... :)

Reply:
Yes, it doesn't seem possible to change those - I have tried !